Essentials/UI/CooldownButton.cs

```diff
@@ -1,295 +1,226 @@
-        public ButtonCategory Category;
-        //private float PixelsPerUnit;
-        private bool CanUse;
-        private Action OnEffectEnd; // rework to events
-        private bool IsDisposed;
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        private CooldownButton(Assembly asm, Action onClick, float cooldown, string embeddedResourcePath/*, float pixelsPerUnit*/, Vector2 positionOffset, ButtonCategory category, bool hasEffectDuration, float effectDuration, Action onEffectEnd)
+        public float MaxTimer = 0f;
+        public float Timer = 0f;
+        public float EffectDuration = 0f;
+        public bool isEffectActive;
+        public bool hasEffectDuration;
+        public bool enabled = true;
+        public UseTester useTester;
+        private byte[] image;
+        private Action OnClick;
+        private Action OnEffectEnd;
+        private HudManager hudManager;
+        private bool canUse;
+
+        public CooldownButton(Action OnClick, float Cooldown, byte[] image, Vector2 PositionOffset, UseTester useTester, HudManager hudManager, float EffectDuration, Action OnEffectEnd)
         {
-            OnEffectEnd = onEffectEnd;
-            PositionOffset = positionOffset;
-            EffectDuration = effectDuration;
-            Category = category;
-            //PixelsPerUnit = pixelsPerUnit;
-            MaxTimer = cooldown;
+            this.hudManager = hudManager;
+            this.OnClick = OnClick;
+            this.OnEffectEnd = OnEffectEnd;
+            this.PositionOffset = PositionOffset;
+            this.EffectDuration = EffectDuration;
+            this.useTester = useTester;
+            MaxTimer = Cooldown;
             Timer = MaxTimer;
+            this.image = image;
+            hasEffectDuration = true;
+            isEffectActive = false;
+            buttons.Add(this);
+            Start();
+        }
 
-            Buttons.Add(this);
-
-            string embeddedResourceFullPath = asm.GetManifestResourceNames().FirstOrDefault(resourceName => resourceName.EndsWith(embeddedResourcePath, StringComparison.Ordinal));
-
-            if (string.IsNullOrEmpty(embeddedResourceFullPath)) throw new ArgumentNullException(nameof(embeddedResourcePath), $"The embedded resource \"{embeddedResourcePath}\" was not found in assembly \"{asm.GetName().Name}\".");
-
-            Stream resourceStream = asm.GetManifestResourceStream(embeddedResourceFullPath);
-
-            KillButtonManager = Object.Instantiate(HudManager.Instance.KillButton, HudManager.Instance.transform);
-
-            //StartColorButton = killButtonManager.renderer.color;
-            StartColorText = KillButtonManager.TimerText.Color;
-
-            KillButtonManager.gameObject.SetActive(true);
-            KillButtonManager.renderer.enabled = true;
+        public CooldownButton(Action OnClick, float Cooldown, byte[] image, Vector2 PositionOffset, UseTester useTester, HudManager hudManager)
```

Comment:
You can use `: this(...)` to call a different constructor, there's no need to the constructor code.